### PR TITLE
Take care of file rename operations in `rename` refactorings

### PR DESCRIPTION
**Pull request by mlangc**
Note that this pull request depends on https://github.com/scala-ide/scala-refactoring/pull/80 and will therefore **not compile** unless an updated version of scala-refactoring becomes available.

Fixes [#1001928](https://www.assembla.com/spaces/scala-ide/tickets/1001928#/activity/ticket:)

### CODE REVIEW

**Comment by sschaef** [comment]
add to whitelist

**Comment by mlangc** [comment]
@sschaef I can update the commit message to reference the associated refactoring commit for further information as soon as we agreed on the latter.

File: org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/ScalaIdeRefactoring.scala [hunk 1]
```diff
@@ -137,9 +145,17 @@ abstract class ScalaIdeRefactoring(val getName: String, val file: ScalaSourceFil
           val msg = "Could not find the corresponding IFile for "+ file.path
           throw new CoreException(new Status(IStatus.ERROR, SdtConstants.PluginId, 0, msg, null))
         }
+    } ++ renameChanges.map { r =>
+      new RenameResourceChange(toIPath(r.sourceFile), r.to)
     }
   }
 
+  private def toIPath(path: AbstractFile) = {
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/refactoring/internal/ScalaIdeRefactoring.scala]: this functionality already exists in `FileUtils`

**Comment by mlangc** [comment]
@sschaef I've moved `toIPath` to `FileUtils` and also updated the commit message to reference scala-ide/scala-refactoring@21e3ccf59123824a22b6d60e4bb991ccc58d2de9.

**Comment by sschaef** [comment]
retest this please

**Comment by sschaef** [comment]
LGTM

**Comment by sschaef** [comment]
Actually, the refactoring on the type is only successful when all the relevant files are opened. Closed files are not changed. But this is a problem unrelated to this PR and just another bug in the refactoring logic.

**Pull request merged**